### ISSUE TITLE
Some questions about s2anet in other dataset

### ISSUE THREAD

**Issue by wangjue-wzq** [csuhan/s2anet#33]
S2anet has good performance in DOTA and HRSC, but when I use s2anet in other data sets, such as DIOR, it has many Fasle Positives. The loss is about 0.3-0.4 and cannot be reduced. Do you have any suggestions?
The config file

`
model = dict(
    type='S2ANetDetector',
    pretrained='torchvision://resnet101',
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs=True,
        num_outs=5),
    rbox_head=dict(
        type='S2ANetHead',
        num_classes=6,
        in_channels=256,
        feat_channels=256,
        stacked_convs=2,
        align_conv_type='AlignConv',#[AlignConv,DCN,GA_DCN]
        align_conv_size=3,
        with_orconv=True,
        anchor_ratios=[1.0],
        anchor_strides=[8, 16, 32, 64, 128],
        anchor_scales=[4],
        target_means=[.0, .0, .0, .0, .0],
        target_stds=[1.0, 1.0, 1.0, 1.0, 1.0],
        loss_fam_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_fam_bbox=dict(
            type='SmoothL1Loss', beta=1.0 / 9.0, loss_weight=1.0),
        loss_odm_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_odm_bbox=dict(
            type='SmoothL1Loss', beta=1.0 / 9.0, loss_weight=1.0)))

train_cfg = dict(
    fam_cfg=dict(
        anchor_target_type='hbb_obb_rbox_overlap',
        assigner=dict(
            type='MaxIoUAssignerRbox',
            pos_iou_thr=0.5,
            neg_iou_thr=0.4,
            min_pos_iou=0,
            ignore_iof_thr=-1),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    odm_cfg=dict(
        anchor_target_type='obb_obb_rbox_overlap',
        anchor_inside_type='center',
        assigner=dict(
            type='MaxIoUAssignerRbox',
            pos_iou_thr=0.5,
            neg_iou_thr=0.4,
            min_pos_iou=0,
            ignore_iof_thr=-1),
        allowed_border=-1,
        pos_weight=-1,
        debug=False))
test_cfg = dict(nms_pre=2000, min_bbox_size=0, score_thr=0.1, nms=dict(type='nms_rotated', iou_thr=0.1), max_per_img=2000)
dataset_type = 'DIORDataset'
data_root = 'data/contest/'
img_norm_cfg = dict(
    mean=[65.288, 76.517, 74.574], std=[54.247, 48.163, 46.390], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='RotatedResize', img_scale=(1024, 1024), keep_ratio=True),
    dict(type='RotatedRandomFlip', flip_ratio=0.5),
    dict(type='RandomRotate', rate=0.5, angles=[30, 60, 90, 120, 150], auto_bound=False),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1024, 1024),
        flip=False,
        transforms=[
            dict(type='RotatedResize', img_scale=(1024, 1024), keep_ratio=True),
            dict(type='RotatedRandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    imgs_per_gpu=4,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        ann_file=data_root + 'trainval_patch_s2a.json',
        img_prefix=data_root + '/images',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=data_root + 'trainval_patch_s2a.json',
        img_prefix=data_root + '/images',
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=data_root + 'test.json',
        img_prefix=data_root + '/images',
        pipeline=test_pipeline))
# optimizer
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[40, 45])
checkpoint_config = dict(interval=1)
# yapf:disable
log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        # dict(type='TensorboardLoggerHook')
    ])
# yapf:enable
# runtime settings
total_epochs = 50
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './works_dirs/s2anet_r101_dota/'
load_from = '.cache/torch/checkpoints/s2anet_r101_fpn_1x_ms_rotate_epoch_12_20200815.pth'
resume_from = None
workflow = [('train', 1)]
`
It trained in 50 epoch
`
{"mode": "train", "epoch": 50, "iter": 50, "lr": 0.0001, "time": 0.86242, "data_time": 0.02099, "memory": 8990, "loss_fam_cls": 0.15524, "loss_fam_bbox": 0.07553, "loss_odm_cls": 0.04947, "loss_odm_bbox": 0.0558, "loss": 0.33603}
{"mode": "train", "epoch": 50, "iter": 100, "lr": 0.0001, "time": 0.8468, "data_time": 0.01285, "memory": 8990, "loss_fam_cls": 0.15316, "loss_fam_bbox": 0.06564, "loss_odm_cls": 0.04806, "loss_odm_bbox": 0.05529, "loss": 0.32216}
{"mode": "train", "epoch": 50, "iter": 150, "lr": 0.0001, "time": 0.85156, "data_time": 0.0137, "memory": 8990, "loss_fam_cls": 0.15407, "loss_fam_bbox": 0.07925, "loss_odm_cls": 0.04821, "loss_odm_bbox": 0.05632, "loss": 0.33786}
{"mode": "train", "epoch": 50, "iter": 200, "lr": 0.0001, "time": 0.85856, "data_time": 0.01323, "memory": 8990, "loss_fam_cls": 0.14702, "loss_fam_bbox": 0.07098, "loss_odm_cls": 0.04571, "loss_odm_bbox": 0.05816, "loss": 0.32187}
{"mode": "train", "epoch": 50, "iter": 250, "lr": 0.0001, "time": 0.84088, "data_time": 0.01274, "memory": 8990, "loss_fam_cls": 0.16415, "loss_fam_bbox": 0.06529, "loss_odm_cls": 0.05354, "loss_odm_bbox": 0.05491, "loss": 0.33789}
{"mode": "train", "epoch": 50, "iter": 300, "lr": 0.0001, "time": 0.83431, "data_time": 0.01131, "memory": 8990, "loss_fam_cls": 0.16803, "loss_fam_bbox": 0.09001, "loss_odm_cls": 0.05105, "loss_odm_bbox": 0.05975, "loss": 0.36885}
{"mode": "train", "epoch": 50, "iter": 350, "lr": 0.0001, "time": 0.84979, "data_time": 0.01281, "memory": 8990, "loss_fam_cls": 0.14657, "loss_fam_bbox": 0.07459, "loss_odm_cls": 0.04058, "loss_odm_bbox": 0.05461, "loss": 0.31635}
{"mode": "train", "epoch": 50, "iter": 400, "lr": 0.0001, "time": 0.85282, "data_time": 0.01256, "memory": 8990, "loss_fam_cls": 0.14932, "loss_fam_bbox": 0.08008, "loss_odm_cls": 0.04741, "loss_odm_bbox": 0.06068, "loss": 0.33749}
{"mode": "train", "epoch": 50, "iter": 450, "lr": 0.0001, "time": 0.84619, "data_time": 0.01359, "memory": 8990, "loss_fam_cls": 0.14492, "loss_fam_bbox": 0.06747, "loss_odm_cls": 0.05152, "loss_odm_bbox": 0.056, "loss": 0.3199}
{"mode": "train", "epoch": 50, "iter": 500, "lr": 0.0001, "time": 0.84169, "data_time": 0.01221, "memory": 8990, "loss_fam_cls": 0.14924, "loss_fam_bbox": 0.06251, "loss_odm_cls": 0.04694, "loss_odm_bbox": 0.04926, "loss": 0.30795}
{"mode": "train", "epoch": 50, "iter": 550, "lr": 0.0001, "time": 0.84714, "data_time": 0.01283, "memory": 8990, "loss_fam_cls": 0.16291, "loss_fam_bbox": 0.08238, "loss_odm_cls": 0.05405, "loss_odm_bbox": 0.05899, "loss": 0.35833}
{"mode": "train", "epoch": 50, "iter": 600, "lr": 0.0001, "time": 0.84862, "data_time": 0.01286, "memory": 8990, "loss_fam_cls": 0.15531, "loss_fam_bbox": 0.06335, "loss_odm_cls": 0.05805, "loss_odm_bbox": 0.05274, "loss": 0.32945}
{"mode": "train", "epoch": 50, "iter": 650, "lr": 0.0001, "time": 0.84977, "data_time": 0.01135, "memory": 8990, "loss_fam_cls": 0.14908, "loss_fam_bbox": 0.07171, "loss_odm_cls": 0.04623, "loss_odm_bbox": 0.05798, "loss": 0.325}

`

**Comment by csuhan** [comment]
1. s2anet performs well on OBB, but we have not validate the HBB performance such as DIOR. The ARF used in ODM may hurt the performance.
2. The loss, `0.3-0.4` is small enough, as you can see `loss_odm_bbox=0.05`. Pay attention to the overfitting problem. You can validate it in epoch 12, 24, 36 etc.
3. You may ned to train your model from ImageNet pretrain, other than DOTA pretrain.

**Comment by wangjue-wzq** [comment]
@csuhan Thanks for your reply. I will try again.
